pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
ok I see, then you should be explicit about testing what you need, e.g. ``ABCInt64Inde``,
 (rather than adding a helper function which serves no other purpose)